### PR TITLE
implement tooling & types

### LINKED ISSUE #213: Create tooling for exporting assets
**Describe what needs to get done**
A tool needs to be made to change layer visibility and export the generated image 

**Describe the solution you'd like**
use psd.js to create a tool

**Additional context**
generating these frames by hand is really not ideal

### PR DESCRIPTION
**Pull request by LoganParker**
Created tooling for generating animation frames. Photoshop uses a frustratingly old version of javascript for scripting, and as such there is no array functionality which makes it quite difficult to read :(

Please check the images, I'm running into the same issue I was running into before with the weird warped bits in the output video and I've forgotten how to fix that, so @edouarde1 if you remember, please let me know! It could also be an issue with the files generated... it's interesting that we're unable to render the rich display on GitHub, which could imply the outputs are corrupted somehow (this looks even more likely considering one of the 32 images is able to be rendered by GitHub...). 

The overlay method I was working on doesn't look super promising, so this is my solution (creating tools so we don't have to do it all manually!!) However, the class which originated from the overlay method is useful for easily updating states and keeping animation logic pretty clean. Avatar was implemented as a class rather than a type as runtime operations are required 

- closes #213
- related to #181 
- related to #192 
- related to #214 

Video output is below:
https://user-images.githubusercontent.com/40530699/215991244-d8fb212e-622a-40c2-b19d-08b0c8d45231.mp4

### CODE REVIEW

**Review by vnajack — APPROVED** [review]
Logan!! Thank you so much for implementing avatar as a class! With the runtime operations required, implementing it as a class is definitely the way to go. There may be items within the avatar class that can be typed, but I'll be focusing on the idle animation and poses so I'm not sure if I can work on it. @mkuelker would you please be able to take this on?

It does seem like there's something going on with the outputs because of the rendering issue.  Is there a way for us to check the files to see if there's anything going on whether it's corruption, metadata, or whatever?

I super appreciate the internal tools to get each frame because doing it manually was very tedious. Well done!

File: speaking-portal-project/src/types.ts [hunk 1]
```diff
@@ -1,14 +1,52 @@
...
 export type MouthCueArray = Array<MouthCue>
 
+// Class instead of type to allow for self referencing & updating during runtime
+export class Avatar {
+    // Readonly because having the avatar body change would be real funky
+    readonly avatar: string
+    blink: boolean
+    phoneme: string
+    arms: string
+    constructor(avatar: string = 'barb', blink: boolean = false, phoneme: string = 'A', arms: string = 'relaxed') {
```
Comment (edouarde1) [inline speaking-portal-project/src/types.ts:21]: I assume we would removed this constructor later on ? Depending on the arguments sent by user

**Review by edouarde1 — CHANGES_REQUESTED** [review]
The script is really cool! Yeah its weird the images are not loading :( 

1) For this PR, with the new images, are they going to break the mp4 files that we are producing currently ? (in terms of the video rendering issues you're having)
- If so maybe we should leave out the photos for now

The class is a great start and we should use that from now on when defining the avatar state. I think as a group we should talk about how we want to represent and simplify the avatar, the avatar states, and the processes we are using the update the avatar states.

File: speaking-portal-project/src/types.ts (hunk 1 above)
Comment (LoganParker) [reply speaking-portal-project/src/types.ts:21]: Those are default values which get overwritten when you pass in arguments to the constructor

**Comment by LoganParker** [comment]
> The script is really cool! Yeah its weird the images are not loading :(
> 
> 1. For this PR, with the new images, are they going to break the mp4 files that we are producing currently ? (in terms of the video rendering issues you're having)

our current outputs work the same as long as you pass in character01 as the avatar. Reminds me I forgot to add the avatar parameter when the avatar is constructed though. Gonna update that as well as the file naming for character01

**Review by edouarde1 — APPROVED** [review]

**Pull request merged**